plugins/hw_event/hw_event_plugin.go

```diff
@@ -0,0 +1,221 @@
+	cneevent "github.com/redhat-cne/sdk-go/pkg/event"
+	"github.com/redhat-cne/sdk-go/pkg/pubsub"
+	"github.com/redhat-cne/sdk-go/pkg/types"
+	v1amqp "github.com/redhat-cne/sdk-go/v1/amqp"
+	v1pubsub "github.com/redhat-cne/sdk-go/v1/pubsub"
+	log "github.com/sirupsen/logrus"
+)
+
+var (
+	resourceAddress string        = "/hw-event"
+	eventInterval   time.Duration = time.Second * 5
+
+	// TODO : pass these from flag.StringVar in main.go
+	apiPath      string   = "/api/cloudNotifications/v1/"
+	apiAddr      string   = "localhost:8080"
+	hwSubUrl     string   = "https://10.19.109.249/redfish/v1/EventService/Subscriptions/"
+	hwEventTypes []string = []string{"StatusChange", "ResourceUpdated", "ResourceAdded", "ResourceRemoved", "Alert"}
+	pub          pubsub.PubSub
+)
+
+type HttpHeader struct {
+	ContentType     string `json:"Content-Type" example:"application/json"`
+	ContentLanguage string `json:"Content-Language" example:"en-US"`
+}
+
+type HwSubPayload struct {
+	Protocol    string       `json:"Protocol" example:"Redfish"`
+	Context     string       `json:"Context"`
+	Destination string       `json:"Destination" example:"https://www.hw-event-proxy-host.com/webhook"`
+	EventTypes  []string     `json:"EventTypes" example:["StatusChange", "ResourceUpdated", "ResourceAdded", "ResourceRemoved", "Alert"]`
+	HttpHeaders []HttpHeader `json:"HttpHeaders"`
+}
+
+type RedfishEvent struct {
+	OdataContext string `json:"@odata.context"`
+	OdataType    string `json:"@odata.type"`
+	Events       []struct {
+		Eventid        string    `json:"EventId"`
+		Eventtimestamp time.Time `json:"EventTimestamp"`
+		Eventtype      string    `json:"EventType"`
+		Memberid       string    `json:"MemberId"`
+		Message        string    `json:"Message"`
+		Messageargs    []string  `json:"MessageArgs"`
+		Messageid      string    `json:"MessageId"`
+		Oem            struct {
+			Hpe struct {
+				OdataContext string `json:"@odata.context"`
+				OdataType    string `json:"@odata.type"`
+				Resource     string `json:"Resource"`
+			} `json:"Hpe"`
+		} `json:"Oem"`
+		Originofcondition string `json:"OriginOfCondition"`
+		Severity          string `json:"Severity"`
+	} `json:"Events"`
+	Name string `json:"Name"`
+}
+
+// Start hw event plugin to process events,metrics and status, expects rest api available to create publisher and subscriptions
+func Start(wg *sync.WaitGroup, scConfig *common.SCConfiguration, fn func(e cneevent.Event) error) error { //nolint:deadcode,unused
+
+	webhookUrl := getWebhookAddr()
+	status, _ := createHwEventSubscription(hwEventTypes, webhookUrl)
+	if status != http.StatusCreated {
+		err := fmt.Errorf("hw event subscription creation api at %s, returned status %d", hwSubUrl, status)
+		return err
+	}
+
+	//channel for the transport handler subscribed to get and set events
+	//eventInCh := make(chan *channel.DataChan, 10)
+	pubSubInstance := v1pubsub.GetAPIInstance(".")
+	endpointURL := &types.URI{URL: url.URL{Scheme: "http", Host: apiAddr, Path: apiPath}}
```

Comment:
config.baseUrl for an example
url := fmt.Sprintf("%s%s", scConfig.BaseURL.String(), "create/event")